src/proxy/inbound.rs

```diff
@@ -113,7 +113,18 @@ impl Inbound {
         request_type: InboundConnect,
         addr: SocketAddr,
     ) -> Result<(), std::io::Error> {
-        let stream = TcpStream::connect(addr).await;
+        let stream = {
+            match &request_type {
+                InboundConnect::DirectPath(stream) => {
+                    let org_src = super::get_original_src_from_stream(stream);
+                    super::freebind_connect(org_src, addr).await
+                },
+                Hbone(req) => {
+                    let org_src = super::get_original_src_from_xff(req);
+                    super::freebind_connect(org_src, addr).await
```

Comment:
Curious if we bind to `org_src`, how could the response packet from the endpoint routed back to this ztunnel

Reply:
In redirection part(i.e. iptables configured by isito-cni) take care of it, it will use conntrack mark and policy route to make the rsp back to ztunnel.

Reply:
Donot we need to setup an additional route table?

Reply:
I remember we do that for original src address support in istio.

Reply:
https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/original_src_filter#extra-setup

Reply:
> I remember we do that for original src address support in istio.

Yes, for ambient ztunnel, these works have been covered  by istio-cni && ztunnel init container.
i.e. : 
In host:
```
-A ztunnel-PREROUTING ! -i vethd7b59a4c -m connmark --mark 0x210/0x210 -j MARK --set-xmark 0x40/0x40
-A ztunnel-PREROUTING -m mark --mark 0x40/0x40 -j RETURN
-A ztunnel-PREROUTING ! -s 10.101.3.14/32 -i vethd7b59a4c -j MARK --set-xmark 0x210/0x210  # ip for ztunnel pod ip, veth is the veth name of ztunnel

# ip rule
0:	from all lookup local
100:	from all fwmark 0x200/0x200 goto 32766
101:	from all fwmark 0x100/0x100 lookup 101
**102:	from all fwmark 0x40/0x40 lookup 102**

# ip r show table 102
default via 10.101.3.14 dev vethd7b59a4c onlink 
10.101.3.14 dev vethd7b59a4c scope link
```

In ztunnel:
```
-A PREROUTING ! -d 10.101.3.14/32 -i eth0 -p tcp -j MARK --set-xmark 0x4d3/0xfff

# ip rule 
0:	from all lookup local
20000:	from all fwmark 0x400/0xfff lookup 100
20001:	from all fwmark 0x401/0xfff lookup 101
20002:	from all fwmark 0x402/0xfff lookup 102
20003:	from all fwmark 0x4d3/0xfff lookup 100
32766:	from all lookup main
32767:	from all lookup default
# ip r show table 100
local default dev lo scope host 
```